File: backend/astra_app/src/main/java/com/wenjun/astra_app/model/vo/FitnessGoal.java
package com.wenjun.astra_app.model.vo;

import lombok.Builder;
import lombok.Data;

import java.util.Date;

@Builder
@Data
public class FitnessGoal {
    private String description;

    private Date targetDate;

    private Double targetValue;

    private String title;

    private Double currentValue;

    private String category;
}

File: astra/lib/src/features/fitness/domain/fitness_goal.dart
class FitnessGoal {
  final String category;
  final double currentValue;
  final String description;
  final DateTime targetTime;
  final double targetValue;
  final String title;

  const FitnessGoal({
    required this.currentValue,
    required this.title,
    required this.description,
    required this.category,
    required this.targetTime,
    required this.targetValue,
  });
}
